src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +44,80 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading]);
+
+  const onFileSelect = (e: any) => {
+    const file: File = e.target.files[0];
+    const formData = new FormData();
+    formData.append('file', file);
+    setUploading(true);
+    fetch(`${process.env.REACT_APP_BACKEND_ADDRESS}/image`, {
+      method: 'POST',
+      body: formData,
+    })
+      .then(res => res.json())
+      .then(() => {
+        setUploading(false);
+        setShowToast(true);
+      });
+  };
+
+  const fileSelector: HTMLInputElement = buildFileSelector();
+  fileSelector.onchange = onFileSelect;
+
+  const handleUploadClick = (e: any) => {
+    e.preventDefault();
+    fileSelector.click();
+  };
 
   return (
     <Container>
+      <Row>
+        <Col>
+          <Button
+            variant="secondary"
+            size="lg"
+            className="my-3"
+            onClick={handleUploadClick}
+            block
+          >
+            {isUploading ? (
+              <Spinner
+                as="span"
+                animation="border"
+                role="status"
+                aria-hidden="true"
+              />
+            ) : (
+              'Upload New Image'
+            )}
+          </Button>
+        </Col>
+      </Row>
       <Row>
         {hasError && <div>Error encountered while loading images.</div>}
         {stimuliUrls && generateCards(stimuliUrls)}
       </Row>
+      {showToast ? (
```

Comment:
More concise if you use the form `showToast && (<component JSZ here>)`. That way we can avoid the `: null` from the ternary operator.